pkg/kubecost/asset.go

```diff
@@ -2737,15 +2757,13 @@ func (as *AssetSet) Resolution() time.Duration {
 }
 
 func (as *AssetSet) Set(asset Asset, aggregateBy []string) error {
-	if as.IsEmpty() {
-		as.Lock()
-		as.assets = map[string]Asset{}
-		as.Unlock()
-	}
-
 	as.Lock()
 	defer as.Unlock()
```

Comment:
Instead of:
```
Read.Lock() // IsEmpty
Read.Unlock() // IsEmpty
Write.Lock() // If 
Write.Unlock() // If 
Write.Lock() // func
Write.Unlock() // func
```

Just put everything behind a single Write Lock.